app/src/main/java/com/deneyehayir/deneysiz/scene/search/SearchViewModel.kt

```diff
@@ -0,0 +1,99 @@
+package com.deneyehayir.deneysiz.scene.search
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.deneyehayir.deneysiz.domain.usecase.FetchSearchResultUseCase
+import com.deneyehayir.deneysiz.internal.extension.toErrorContentUiModel
+import com.deneyehayir.deneysiz.scene.search.model.toUiModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.flow.debounce
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class SearchViewModel @Inject constructor(
+    private val fetchSearchResultUseCase: FetchSearchResultUseCase
+) : ViewModel() {
+
+    private val _uiState = MutableStateFlow(SearchUiState.Initial)
+    val uiState: StateFlow<SearchUiState> get() = _uiState
+
+    var queryFlowText = MutableStateFlow("")
+        private set
+
+    fun handleUiEvents(event: SearchUiEvents) {
+        when (event) {
+            is SearchUiEvents.QueryTextChange -> {
+                queryFlowText.value = event.query
+                makeSearch()
+            }
+            SearchUiEvents.CancelButtonClick -> {
+                _uiState.update {
+                    it.cancelActions()
+                }
+                queryFlowText.value = ""
+            }
+            SearchUiEvents.ErrorCloseClick -> {
+                _uiState.update {
+                    it.hideError()
+                }
+            }
+            SearchUiEvents.ErrorRetryClick -> {
+                _uiState.update {
+                    it.hideError()
+                }
+                makeSearch()
+            }
+        }
+    }
+
+    @OptIn(FlowPreview::class, ExperimentalCoroutinesApi::class)
```

Comment:
Instead of adding these annotations for each function, we can add them as `freeCompilerArgs`.